src/Symfony/Bridge/PhpUnit/ErrorAssert.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit;
+
+/**
+ * Test that your code triggers expected error messages.
+ *
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+final class ErrorAssert
+{
+    public static function assertDeprecationsAreTriggered($expectedMessages, callable $testCode)
+    {
+        self::assertErrorsAreTriggered(E_USER_DEPRECATED, $expectedMessages, $testCode);
+    }
+
+    public static function assertErrorsAreTriggered($expectedType, $expectedMessages, callable $testCode)
+    {
+        $triggeredMessages = array();
+
+        try {
+            $prevHandler = set_error_handler(function ($type, $message, $file, $line, $context) use ($expectedType, &$triggeredMessages, &$prevHandler) {
+                if ($expectedType !== $type) {
+                    return null !== $prevHandler && call_user_func($prevHandler, $type, $message, $file, $line, $context);
+                }
+                $triggeredMessages[] = $message;
+            });
+
+            $testCode();
```

Comment:
@Ener-Getick But that would mean to drop support for other callables that are not anonymous functions.